### ISSUE TITLE
Esc key behavior for 2nd-level modals

### ISSUE THREAD

**Issue by kevinwhoffman** [WordImpress/google-maps-builder#82]
Pressing the Esc key inside the map builder closes the map builder modal, which makes sense. However, there are several instances where the UI opens a 2nd-level modal within the map builder modal:

* Add Location
* Goto Location
* Edit Map Title
* Customize Map Marker
* Upload Marker

When pressing the Esc key in these 2nd-level modals, the entire map builder closes, which is a frustrating behavior. The expected behavior is to have the 2nd-level modal close, but the map builder itself remain open.

**Comment by mathetos** [comment]
Not sure whether that can be separated out so easily. But there is an "X" on those second level modals as well.

@DevinWalker  thoughts?

**Comment by kevinwhoffman** [comment]
Yeah, I realize the X, but I've done this over and over since I started testing. One reason is that I often click the Add Location button when I mean to click Drop a Marker, and I quickly try to Esc out of it but instead end up closing the whole map builder.

**Comment by DevinWalker** [comment]
It wouldn't be too hard to implement this... good issue :+1:

**Comment by nishitlangaliya** [comment]
@kevinwhoffman , @DevinWalker ,

As per suggestions I have fixed the escape key behaviour for 2nd level modal popup and made PR https://github.com/WordImpress/maps-builder-core/pull/56 for the same.

Video:

https://www.useloom.com/share/a4d38c6bbb2141f88a8d09b74861b8ca

Thanks
